GVFS/GVFS.Common/InstallerPreRunChecker.cs

```diff
@@ -147,6 +147,16 @@ protected virtual bool IsProjFSInboxed()
             return GVFSPlatform.Instance.KernelDriver.IsGVFSUpgradeSupported();
         }
 
+        protected virtual bool FreshInstallOrServiceRunning()
+        {
+            bool isInstalled;
+            bool isRunning;
+
+            GVFSPlatform.Instance.IsServiceInstalledAndRunning(GVFSConstants.Service.ServiceName, out isInstalled, out isRunning);
+
+            return isInstalled ? isRunning : true;

```

Comment:
This is essentially "isInstalled implies isRunning" which can be given as `!isInstalled || isRunning` (which also seems to match your function name a bit better).